### PR TITLE
feat: game over if snake touches the wall

### LINKED ISSUE #86: kill the snakes on the wall
For now the snake can move through walls. Lets make the game ends if the head of the snake hit the wall.

### PR DESCRIPTION
**Pull request by yvisfreire**
Adds game over if the snake touches the game map boundaries.

Closes #86

### CODE REVIEW

**Comment by rzimmerdev** [comment]
It would be nice to have the game automatically restart after the snake dies!

**Comment by rzimmerdev** [comment]
I've refactored the code to be modularized and added the restart condition.

**Review by VictorRS27 — APPROVED** [review]

**Review by vrrusso — CHANGES_REQUESTED** [review]
For now tha game instantly restarts on ending and the timer is not turned to zero. It would be nice if after dying, the enlapsed time is shown, and the timer resets

File: src/snaskii.c [hunk 1]
```diff
@@ -104,6 +104,11 @@ void play_game(scene_t scenes[GAME_SCENES_SIZE], times_t* times)
 		// TODO: Advance game
 		move_snake(&snake);
 
+		if (!snake.alive) {
+            coord_t energy_blocks[ENERGY_BLOCKS_SIZE];
+            init_game(&snake, energy_blocks);
+		}
```
Comment (leofusineli) [inline src/snaskii.c:110]: if the snake is not alive does the game display some message informing the snake died or does it just restarts itself?

Comment (jj-marinho) [reply src/snaskii.c:110]: Currently it just restarts. I'm working on a death screen for this situation right now

Comment (leofusineli) [reply src/snaskii.c:110]: Great

**Review by leofusineli — APPROVED** [review]

**Pull request merged**